Package.swift

```diff
@@ -81,5 +92,40 @@ let package = Package(
                 .process("Fixtures"),
             ]
         ),
+        .testTarget(
+            name: "MapboxNavigationTests",
+            dependencies: [
+                "MapboxNavigation",
+                "FBSnapshotTestCase",
+                "Quick",
+                "TestHelper",
+                "OHHTTPStubs",
+                "CarPlayTestHelper",
+            ],
+            exclude: [
+                // Exclude snapshot testing tests because it is not clear how to configure FBSnapshotTestCase in SPM.
+                "Info.plist",
+                "MapboxNavigationTests-Bridging.h",
+                "BottomBannerSnapshotTests.swift",
+                "CarPlayCompassViewSnapshotTests.swift",
+                "GuidanceCardsSnapshotTests.swift",
+                "SnapshotTest+Mapbox.swift",
+                "NavigationViewControllerTests.swift",
+                "CarPlayManagerTests.swift", // There are issues with setting accessToken
```

Comment:
Is the issue that SPM runs tests without a bundle that includes Info.plist? In #2876, @ShanMa1991 worked around that issue by having Mapbox(Core)Navigation look for a key in UserDefaults as an optional alternative to Info.plist; the user default would only be set inside the test. (#2911 implemented a more robust solution for Info.plist keys that are needed during production, but that probably isn’t necessary here.)

Reply:
I think I checked the stack trace and there wasn't a way to provide this access token at that time. I will get back to this after the chain of build system PRs will be merged.

Reply:
So here is what happens:

`CarPlayMapViewController.setupPassiveLocationManager()` creates `PassiveLocationDataSource` which in turn accesses `Directions.shared`. `Directions.shared` is initialized with the `DirectionsCredentials()` which looks for access token only in Info.plist.

Reply:
> `CarPlayMapViewController.setupPassiveLocationManager()` creates `PassiveLocationDataSource` which in turn accesses `Directions.shared`.

It’s potentially problematic that `Directions.shared` is used in this case. What if a developer needs to specify an access token programmatically rather than in Info.plist? Most other classes have a way to specify the credentials explicitly on initialization.

Reply:
I'm happy to address this concern as a new issue.